### PR TITLE
Add auxilia operator

### PR DESCRIPTION
**Pull request by shirly121**
## Related issue number

Fixes

### CODE REVIEW

File: research/query_service/ir/compiler/src/test/java/com/alibaba/graphscope/gremlin/ValueMapTest.java [hunk 1]
```diff
@@ -38,7 +39,7 @@
     public void g_V_valueMap_strs_test() {
         Traversal traversal = g.V().valueMap("name", "id");
         Step valueMapStep = traversal.asAdmin().getEndStep();
-        ProjectOp op = (ProjectOp) IrPlanBuidler.StepTransformFactory.VALUE_MAP_STEP.apply(valueMapStep);
+        ProjectOp op = (ProjectOp) StepTransformFactory.VALUE_MAP_STEP.apply(valueMapStep);
 
         List<Pair> expected = Arrays.asList(Pair.with("@.name", ArgUtils.strAsNameId("~@.name")),
```
Comment (longbinlai) [inline research/query_service/ir/compiler/src/test/java/com/alibaba/graphscope/gremlin/ValueMapTest.java:44]: 这个不是 @.~name, 和 @.~id吗

Comment (shirly121) [reply research/query_service/ir/compiler/src/test/java/com/alibaba/graphscope/gremlin/ValueMapTest.java:44]: 系统自动生成的alias会以～作为开头，现在project出来的alias的命名规则就是~+expr

File: research/query_service/ir/compiler/src/main/java/com/alibaba/graphscope/common/intermediate/AliasArg.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
+package com.alibaba.graphscope.common.intermediate;
+
+import com.alibaba.graphscope.common.jna.type.FfiNameOrId;
+import com.google.common.base.Objects;
+
+public class AliasArg {
+    private FfiNameOrId.ByValue alias;
+    private boolean isQueryGiven;
+
```
Comment (shirly121) [inline research/query_service/ir/compiler/src/main/java/com/alibaba/graphscope/common/intermediate/AliasArg.java:9]: add the new class to carry isQueryGiven

File: research/query_service/ir/compiler/src/main/java/com/alibaba/graphscope/common/IrPlan.java [hunk 3]
```diff
@@ -195,10 +203,8 @@ public Pointer apply(InterOpBase baseOp) {
                 Pointer ptrProject = irCoreLib.initProjectOperator(false);
                 exprList.forEach(pair -> {
                     String expr = (String) pair.getValue0();
-                    FfiNameOrId.ByValue alias = (FfiNameOrId.ByValue) pair.getValue1();
-                    // aliases generated by compiler all start with ~
-                    boolean isQueryGiven = !ArgUtils.isHiddenStr(alias.name);
-                    irCoreLib.addProjectExprAlias(ptrProject, expr, alias, isQueryGiven);
+                    AliasArg alias = (AliasArg) pair.getValue1();
+                    irCoreLib.addProjectExprAlias(ptrProject, expr, alias.getAlias(), alias.isQueryGiven());
```
Comment (shirly121) [inline research/query_service/ir/compiler/src/main/java/com/alibaba/graphscope/common/IrPlan.java:207]: get isQueryGiven from AliasArg when setting project operator

File: research/query_service/ir/compiler/src/main/java/com/alibaba/graphscope/common/IrPlan.java [hunk 4]
```diff
@@ -229,6 +235,34 @@ public Pointer apply(InterOpBase baseOp) {
...
+                    propertyList.sort(Comparator.comparing(FfiNameOrId::hashCode));
+                    propertyList.forEach(k -> {
+                        irCoreLib.addAuxiliaProperty(ptrAuxilia, k);
+                    });
+                }
+
+                Optional<OpArg> aliasOpt = baseOp.getAlias();
+                if (aliasOpt.isPresent()) {
+                    AliasArg alias = (AliasArg) aliasOpt.get().getArg();
+                    irCoreLib.setAuxiliaAlias(ptrAuxilia, alias.getAlias(), alias.isQueryGiven());
```
Comment (shirly121) [inline research/query_service/ir/compiler/src/main/java/com/alibaba/graphscope/common/IrPlan.java:262]: get isQueryGiven from AliasArg when setting auxilia operator

File: research/query_service/ir/compiler/src/test/java/com/alibaba/graphscope/common/intermediate/process/AliasTest.java [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+import com.alibaba.graphscope.common.jna.IrCoreLibrary;
+import com.alibaba.graphscope.common.jna.type.FfiNameOrId;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.function.Function;
+
+public class AliasTest {
+    private IrCoreLibrary irCoreLib = IrCoreLibrary.INSTANCE;
```
Comment (shirly121) [inline research/query_service/ir/compiler/src/test/java/com/alibaba/graphscope/common/intermediate/process/AliasTest.java:17]: add AliasTest to test AliasProcessor

File: research/query_service/ir/compiler/src/test/java/com/alibaba/graphscope/common/intermediate/process/PropertyDetailsTest.java [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+import org.javatuples.Pair;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.function.Function;
+
+public class PropertyDetailsTest {
+    private static IrCoreLibrary irCoreLib = IrCoreLibrary.INSTANCE;
```
Comment (shirly121) [inline research/query_service/ir/compiler/src/test/java/com/alibaba/graphscope/common/intermediate/process/PropertyDetailsTest.java:23]: add PropertyDetailsTest to test PropertyDetailsProcessor

File: research/query_service/ir/compiler/src/test/java/com/alibaba/graphscope/gremlin/TraversalPropertiesTest.java [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+import org.apache.tinkerpop.gremlin.process.traversal.Traversal;
+import org.apache.tinkerpop.gremlin.process.traversal.dsl.graph.GraphTraversalSource;
+import org.apache.tinkerpop.gremlin.process.traversal.dsl.graph.__;
+import org.apache.tinkerpop.gremlin.structure.Graph;
+import org.apache.tinkerpop.gremlin.tinkergraph.structure.TinkerFactory;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class TraversalPropertiesTest {
+    private Graph graph = TinkerFactory.createModern();
```
Comment (shirly121) [inline research/query_service/ir/compiler/src/test/java/com/alibaba/graphscope/gremlin/TraversalPropertiesTest.java:19]: add TraversalPropertiesTest to test whether Auxilia operator can be generated properly from a gremlin traversal

**Pull request merged**